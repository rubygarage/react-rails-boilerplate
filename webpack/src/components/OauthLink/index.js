import React from 'react'
import PropTypes from 'prop-types'
import Icon from 'components/ui/Icon'

function OauthLink({ type, onClick }) {
  return (
    <a href="" className="social-round-icon" onClick={onClick}>
      <Icon type={type} />
    </a>
  )
}

OauthLink.propTypes = {
  onClick: PropTypes.func.isRequired,
  type: PropTypes.string.isRequired
}

export default OauthLink
