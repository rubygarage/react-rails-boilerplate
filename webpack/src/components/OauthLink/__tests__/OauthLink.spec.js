import React from 'react';
import { shallow } from 'enzyme';
import OauthLink from 'components/OauthLink';

describe('<OauthLink />', () => {
  const defaultProps = {
    onClick: jest.fn(),
    type: 'icon-fb',
  };

  it('renders oauth link', () => {
    const component = shallow(<OauthLink {...defaultProps} />);

    expect(component).toMatchSnapshot();
  });
});
