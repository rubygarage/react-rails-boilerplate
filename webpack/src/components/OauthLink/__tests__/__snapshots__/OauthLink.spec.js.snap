// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OauthLink /> renders oauth link 1`] = `
<a
  className="social-round-icon"
  href=""
  onClick={[Function]}
>
  <Icon
    type="icon-fb"
  />
</a>
`;
